### PR TITLE
[WAL-719] MGI integration example

### PR DESCRIPTION
**Pull request by Ifropc**

### CODE REVIEW

File: examples/MGI/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## MoneyGram integration example
+### Code structure
+This module contains an example for integration with the MoneyGram.
```
Comment (JakeUrban) [inline examples/MGI/README.md:3]:
```suggestion
This module contains an example for integration with MoneyGram.  
```

File: examples/MGI/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+to run either of these services. Pass either `deposit`, `withdrawal`, `transaction` or `info` as 
+an argument to run the desired service.
+
+### Environmental variables
+You will need to define following env variables to successfully run the code:
+1. `CLIENT_DOMAIN` [SEP-10](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md) client domain
+2. `CLIENT_PRIVATE` private key from SEP-1 client domain file. Alternatively, you may want to implement your own `WalletSigner`.
+3. (Optional) `STELLAR_KEY` private key of the user account with established USDC trustline. If not specified, default will be used. (Please do not drain the funds of the example account)
+
+In addition, when calling `trasacntion` service, you will need auth token and transaction id. (Auth token is printed during withdrawal/deposit flow in the console):
```
Comment (JakeUrban) [inline examples/MGI/README.md:20]:
```suggestion
In addition, when calling `transaction` service, you will need auth token and transaction id. (Auth token is printed during withdrawal/deposit flow in the console):
```

File: examples/MGI/src/main/kotlin/org/stellar/example/Deposit.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.delay
+import org.stellar.walletsdk.anchor.TransactionStatus
+
+object Deposit {
+  @JvmStatic
+  suspend fun main() {
+    // Authorizing
+    val token = anchor.auth().authenticate(myAccount, Signer, clientDomain = clientDomain)
```
Comment (JakeUrban) [inline examples/MGI/src/main/kotlin/org/stellar/example/Deposit.kt:11]: Where is `Signer` defined?

Comment (JakeUrban) [reply examples/MGI/src/main/kotlin/org/stellar/example/Deposit.kt:11]: Or is this kotlin's way of passing the default signer?

File: examples/MGI/src/main/kotlin/org/stellar/example/Withdrawal.kt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    // Wait for user input
+    do {
+      // Get transaction info
+      transaction = anchor.getTransaction(withdrawal.id, token)
+      delay(5.seconds)
+    } while (transaction.status != TransactionStatus.PENDING_USER_TRANSFER_START)
+
+    // Send transaction with transfer
+    val transfer = (transaction as WithdrawalTransaction).toTransferTransaction(stellar, asset)
```
Comment (JakeUrban) [inline examples/MGI/src/main/kotlin/org/stellar/example/Withdrawal.kt:40]: I still don't love the "transfer" name for stellar transactions. Maybe we can rename to `StellarTransaction`? Or rename to `Transaction` and rename the existing class to `AnchorTransaction`? Not urgent.

**Review by JakeUrban — APPROVED** [review]

File: examples/MGI/src/main/kotlin/org/stellar/example/Deposit.kt (hunk 3 above)
Comment (Ifropc) [reply examples/MGI/src/main/kotlin/org/stellar/example/Deposit.kt:11]: It's defined [here](https://github.com/stellar/kotlin-wallet-sdk/blob/36bb82fffaab2064af08216c532a2363434e315d/examples/MGI/src/main/kotlin/org/stellar/example/Withdrawal.kt#L63), I will move it into it's own file

File: examples/MGI/src/main/kotlin/org/stellar/example/Withdrawal.kt (hunk 4 above)
Comment (Ifropc) [reply examples/MGI/src/main/kotlin/org/stellar/example/Withdrawal.kt:40]: Transactions coming from anchors are already named `AnchorTransaction`
https://github.com/stellar/kotlin-wallet-sdk/blob/main/wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/AnchorTransaction.kt
`.toTransferTransaction` is a helper function that transforms existing `AnchorTransaction` to Stellar transaction with one Transfer operation inside. 
We can rename it to `toStellarTransfer`, WDYT?

**Pull request merged**